### PR TITLE
chore: update docs around redirect()

### LINKED ISSUE #61315: Docs: Is that in the wrong place?
### What is the improvement or update you wish to see?

Move `redirect` related "Good to know" item from "`useRouter` hook" to "redirect function" section.

### Is there any context that might help us understand?

Why mentioned `redirect` in "`useRouter` hook" section? It's confusing. Should that be put in "redirect function" section? 

Xnip2024-01-29_18-07-07

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/building-your-application/routing/redirecting#userouter-hook

### PR DESCRIPTION
**Pull request by samcx**
## Description
- Add a sentence for clarifying redirecting in a Server Component
- Fixes https://github.com/vercel/next.js/issues/61315

Closes NEXT-2284

### CODE REVIEW

File: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx [hunk 1]
```diff
@@ -184,6 +184,58 @@ For a full list of `useRouter` methods, see the [API reference](/docs/app/api-re
 
 > **Recommendation:** Use the `<Link>` component to navigate between routes unless you have a specific requirement for using `useRouter`.
 
+## `redirect` function
+
+For [Server Components](/docs/app/building-your-application/rendering/server-components), use the `redirect` function function instead.
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:189]:
```suggestion
For [Server Components](/docs/app/building-your-application/rendering/server-components), use the `redirect` function instead.
```

File: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx [hunk 2]
```diff
@@ -184,6 +184,58 @@ For a full list of `useRouter` methods, see the [API reference](/docs/app/api-re
 
 > **Recommendation:** Use the `<Link>` component to navigate between routes unless you have a specific requirement for using `useRouter`.
 
+## `redirect` function
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:187]:
```suggestion
## `redirect` Function
```

File: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx [hunk 3]
```diff
@@ -184,6 +184,58 @@ For a full list of `useRouter` methods, see the [API reference](/docs/app/api-re
...
+
+For [Server Components](/docs/app/building-your-application/rendering/server-components), use the `redirect` function function instead.
+
+```tsx filename="app/team/[id]/page.js"
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:191]:
```suggestion
```tsx filename="app/team/[id]/page.tsx" switcher
```

File: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx [hunk 4]
```diff
@@ -184,6 +184,58 @@ For a full list of `useRouter` methods, see the [API reference](/docs/app/api-re
...
+  const team = await fetchTeam(params.id)
+  if (!team) {
+    redirect('/login')
+  }
+
+  // ...
+}
+```
+
+```jsx filename="app/team/[id]/page.js"
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:210]:
```suggestion
```jsx filename="app/team/[id]/page.js" switcher
```

File: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx [hunk 5]
```diff
@@ -184,6 +184,58 @@ For a full list of `useRouter` methods, see the [API reference](/docs/app/api-re
...
+  }
+
+  // ...
+}
+```
+
+> **Good to know**:
+>
+> - `redirect` returns a 307 (Temporary Redirect) status code by default. When used in a Server Action, it returns a 303 (See Other), which is commonly used for redirecting to a success page as a result of a POST request.
+> - `redirect` internally throws an error so should be called outside of `try/catch` blocks.
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:232]:
```suggestion
> - `redirect` internally throws an error so it should be called outside of `try/catch` blocks.
```

File: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx (hunk 2 above)
Comment (samcx) [reply docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:187]: @ztanner It was lowercase elsewhere, so I'll fix that too then

Comment (samcx) [reply docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx]: Actually, it's lowercase in most places <picture data-single-emoji=":thinkies:" title=":thinkies:">:thinkies:</picture>

File: docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx [hunk 6]
```diff
@@ -35,7 +35,7 @@ There are a few ways you can handle redirects in Next.js. This page will go thro
 
 <AppOnly>
 
-## `redirect` function
+## `redirect` Function
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx:38]:
```suggestion
## `redirect` function
```

**Review by ztanner — APPROVED** [review]

**Pull request merged**